cmd/buildkitd/main.go

```diff
@@ -212,7 +219,40 @@ func serveGRPC(server *grpc.Server, addrs []string, errCh chan error) error {
 	return nil
 }
 
-func getListener(addr string) (net.Listener, error) {
+// Convert a string containing either a group name or a stringified gid into a numeric id)
+func groupToGid(group string) (int, error) {
+	if group == "" {
+		return os.Getgid(), nil
+	}
+
+	var (
+		err error
+		id  int
+	)
+
+	// Try and parse as a number, if the error is ErrSyntax
+	// (i.e. its not a number) then we carry on and try it as a
+	// name.
+	if id, err = strconv.Atoi(group); err == nil {
+		return id, nil
+	} else if err.(*strconv.NumError).Err != strconv.ErrSyntax {
```

Comment:
Is this cast to `strconv.NumError` safe? Goot to me to just do `LookupGroup` for any error.

Reply:
`[strconv.Atoi](https://golang.org/pkg/strconv/#Atoi)` docs say this is just a call to `[strconv.ParseInt](https://golang.org/pkg/strconv/#ParseInt)` whose docs say "The errors that ParseInt returns have concrete type `*NumError`" so I think the cast is always safe.

The other main error other than `ErrSyntax` is `ErrRange` and I wanted to prevent the possibility of trying to looking up `999999999999999999999999` as a group name since all numeric groups are not (AIUI) allowed. TBH the only real downside of using `LookupGroup` unconditionally is that a `number too big` error message becomes a `group not found` error message which isn't entirely accurate, or worst case I suppose it could unexpectedly succeed (and then perhaps error on the next `Atoi`, quite why `Group.Gid` is a string I've no idea....) Maybe none of that is a big deal though.